cfme/infrastructure/virtual_machines.py

```diff
@@ -553,22 +571,23 @@ class InfraVm(VM):
         template_name: Name of the template to use for provisioning
     """
 
+    @attr.s
+    # TODO snapshot collections
     class Snapshot(object):
         snapshot_tree = ManageIQTree('snapshot_treebox')
 
-        def __init__(self, name=None, description=None, memory=None, parent_vm=None):
-            super(InfraVm.Snapshot, self).__init__()
-            self.name = name
-            self.description = description
-            self.memory = memory
-            self.vm = parent_vm
+        name = attr.ib(default=None)
+        description = attr.ib(default=None)
+        memory = attr.ib(default=None)
+        parent_vm = attr.ib(default=None)
 
         @property
         def exists(self):
-            title = self.description if self.vm.provider.one_of(RHEVMProvider) else self.name
-            view = navigate_to(self.vm, 'SnapshotsAll')
+            from cfme.infrastructure.provider.rhevm import RHEVMProvider
```

Comment:
That would force `exists` to be a method and not a property, which I'd like to avoid.  I'm seeing a second use in `test_create_snapshot_via_ae` as well, but that's reasonable to update as well.

I've got an idea for this, which I'm submitting now as commit:

Set `InfraProvider.SNAPSHOT_TITLE = 'name'` and `RHEVMProvider.SNAPSHOT_TITLE = 'description'` and then do a `getattr()` here using that provider constant.

Reply:
I mean when Instantiating Snapshot, the 'title' kwarg is passed in by the instantiator and it sets `self.title` in Snapshot. Then 'exists' property uses self.title

Reply:
Your new approach is good too.